src/test/java/com/google/sps/GetPlayerNameFromDatabaseTest.java

```diff
@@ -0,0 +1,99 @@
+public final class GetPlayerNameFromDatabaseTest {
+  // Mock current user for testing
+  private static final String NAME = "alex";
+  private static final String EMAIL = "alex@email.com";
+  private static final String AUTH_DOMAIN = "email.com";
+  private static final String CURR_USER_ID = "testid";
+  private static final String IMAGE_ID = "imageId";
+  private static final String GAME_STAGE_ID = "gameStageId";
+  private static final String HAT_ID = "hat";
+  private static final String GLASSES_ID = "glasses";
+  private static final String COMPANION_ID = "companion";
+  private static final List<String> ALL_ACCESSORIES =
+      Arrays.asList(HAT_ID, GLASSES_ID, COMPANION_ID);
+  private static final int NO_EXPERIENCE = 0;
+  private static final int THRESHOLD_10 = 10;
+  private static final int THRESHOLD_15 = 15;
+  private static Map<String, Object> USER_ID_CONFIG = new HashMap<>();
+
+  static {
+    USER_ID_CONFIG.put("com.google.appengine.api.users.UserService.user_id_key", CURR_USER_ID);
+  }
+
+  // creating a test user service, setting current user to be the above mock
+  private final LocalUserServiceTestConfig localUserServiceTestConfig =
+      new LocalUserServiceTestConfig().setOAuthUserId(CURR_USER_ID);
+  private final LocalDatastoreServiceTestConfig localDatastoreServiceTestConfig =
+      new LocalDatastoreServiceTestConfig();
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(localUserServiceTestConfig, localDatastoreServiceTestConfig)
+          .setEnvIsAdmin(true)
+          .setEnvAuthDomain(AUTH_DOMAIN)
+          .setEnvEmail(EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAttributes(USER_ID_CONFIG);
+  private PlayerDatabase playerDatabase;
+  private static final Gson gson = new Gson();
+
+  @Before
+  public void setUp() {
+    helper.setUp(); // initialize local datastore for testing
+    DatastoreService localDatastore = DatastoreServiceFactory.getDatastoreService();
+    UserService localUserService = UserServiceFactory.getUserService();
+    this.playerDatabase = new PlayerDatabase(localDatastore, localUserService);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /** Tests that the player's name is queried back sucessfully */
+  @Test
+  public void getPlayerNameFromDatabase_worksSuccessfully() throws LoggedOutException {
+    Player player =
+        new Player(
+            NAME,
+            EMAIL,
+            CURR_USER_ID,
+            IMAGE_ID,
+            GAME_STAGE_ID,
+            ALL_ACCESSORIES,
+            NO_EXPERIENCE,
+            THRESHOLD_15);
+    player.setEquippedHatID(HAT_ID);
+    player.setEquippedGlassesID(GLASSES_ID);
+    player.setEquippedCompanionID(COMPANION_ID);
+    this.playerDatabase.addPlayerToDatabase(player);
+
+    String resultPlayerName = this.playerDatabase.getEntityDisplayName();
+    String expectedPlayerName = NAME;
+    Assert.assertEquals(resultPlayerName, expectedPlayerName);
```

Comment:
Also testing a second user might (or might not ) break since playerdatabase / datastore are final in your servlet

Reply:
changed that so database is not final